gateleen-core/src/main/java/org/swisspush/gateleen/core/util/ExpansionDeltaUtil.java

```diff
@@ -277,7 +280,23 @@ public static CollectionResourceContainer verifyCollectionResponse(String target
             throw new ResourceCollectionException("Request did not return data. Invalid usage of params " + params + " ?", StatusCode.BAD_REQUEST);
         }
 
-        JsonObject obj = new JsonObject(data.toString("UTF-8"));
+        // Parse payload
+        final String dataAsString = data.toString(UTF_8);
+        final JsonObject obj;
+        try {
+            obj = new JsonObject(dataAsString);
+        } catch (DecodeException e) {
+            final int MAX_CHARS_TO_PRINT = 4096;
+            final String msgHead = "Failed to decode data as JSON.";
+            final String msgTail = "First " + MAX_CHARS_TO_PRINT + " characters were:\n" +
+                    // Append first n characters from malformed data
+                    dataAsString.substring(0, Math.min(MAX_CHARS_TO_PRINT, dataAsString.length())) + "\n";
+            // Log as INFO only because this situation is not our fault and we can continue
+            // our service with no problems.
+            log.info(msgHead, e);
+            throw new ResourceCollectionException(msgHead + "\n\n" + msgTail, StatusCode.BAD_GATEWAY, e);
```

Comment:
Do we really end the response in this case?
A not-ended response is the original issue (i.e. client is waiting forever for the response).
Can you please explain why this new Exception Handler is enough?

Reply:
Yes this suffice :)

Because down the stack in [DeltaHandler.java:255](https://github.com/swisspush/gateleen/blob/v1.1.34/gateleen-delta/src/main/java/org/swisspush/gateleen/delta/DeltaHandler.java#L255) where we do the call, we handle exactly this exception and 'end' the response at [DeltaHandler.java:295](https://github.com/swisspush/gateleen/blob/v1.1.34/gateleen-delta/src/main/java/org/swisspush/gateleen/delta/DeltaHandler.java#L295).

You can observe, that this only adapts error handling like other cases in same method e.g. [ExpansionDeltaUtil.java:240](https://github.com/swisspush/gateleen/pull/242/files/720ff27433515a76553df3926977eb2407f857a8#diff-7a73c7cbdcc00d2dac0ade9446aa3f85L277)

Reply:
Cool. Thx